swap/src/protocol/alice/state.rs

```diff
@@ -578,6 +617,22 @@ impl State3 {
             .context("Failed to complete Bitcoin punish transaction")
     }
 
+    /// Construct tx_early_refund, sign it with Bob's signature and our own.
+    /// If we do not have a Bob's signature stored, we return None.
+    pub fn signed_early_refund_transaction(&self) -> Option<Result<bitcoin::Transaction>> {
```

Comment:
just a feeling, but Result<Option<Transaction>> feels like a more intuitive api

Reply:
Doesn't really make sense though because if it's None it won't ever fail. If we made this Bone we'll have to check for errors more often than we need to.